src/include/module/ipc4/base-config.h

```diff
@@ -182,7 +182,7 @@ enum ipc4_base_module_params {
 
 struct ipc4_pin_props {
 	/* type of the connected stream. */
-	enum ipc4_stream_type stream_type;
```

Comment:
About this structure: my initial thought was that this is used in IPC4 communication between Linux and SOF, but after looking a bit more carefully it doesn't seem to be the case.

From this comment:

```
This file contains structures that are exact copies of an existing ABI used
by IOT middleware. They are Intel specific and will be used by one middleware.

Some of the structures may contain programming implementations that makes them
unsuitable for generic use and general usage.

This code is mostly copied "as-is" from existing C++ interface files hence the use of
different style in places. The intention is to keep the interface as close as possible to
original so it's easier to track changes with IPC host code.
```

My guess is this is being used by some other entity (?) and it would seem like this is Intel specific.

Since this is also included in files used by IPC3 platforms we get the compilation warning mentioned in https://github.com/thesofproject/sof/issues/9378 (i.e: `warning: 'format' offset 1 in 'struct ipc4_pin_props' isn't aligned to 4`) (caused by the size of `enum ipc4_stream_type`).

With this in mind, I'm not sure the replacement with the `uint32_t` type is right as I don't fully understand how these IPC4 structures are used. Can someone from Intel comment on this, please?

CC: @RanderWang @ranj063

Reply:
I think this is OK now, we were doing a lot of work at that time to ensure compatibility, but that good now.

Reply:
alright, I'm still somewhat skeptical of the IPC4-related change but if ppl from Intel are fine with it then that's fine with me